### ISSUE TITLE
libgit2 and git2r not in good terms

### ISSUE THREAD

**Issue by Anirban166** [Anirban166/Autocomment-atime-results#39]
Current workflow *suddenly* started resulting in:
```r
* installing *source* package ‘git2r’ ...
** package ‘git2r’ successfully unpacked and MD5 sums checked
** using staged installation
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
configure: error: in '/tmp/RtmpaVPwMD/R.INSTALL164a478e4351/git2r':
checking whether the libgit2 version will work in git2r... no
configure: error: package dependency requirement 'libgit2 >= 1.0' could not be satisfied.

See 'config.log' for more details
  -----------------------------------------------------------------------

   Configuration failed to find libgit2 library. Try installing:
     * brew: libgit2 (MacOS)
     * deb: libgit2-dev   (Debian, Ubuntu, etc)
     * rpm: libgit2-devel (Fedora, CentOS, RHEL)
   and try again.

   If the libgit2 library is installed on your system but the git2r
   configuration is unable to find it, you can specify the include and
   lib path to libgit2 with:

   given you downloaded a tar-gz archive:
   R CMD INSTALL git2r-.tar.gz --configure-vars='INCLUDE_DIR=/path/to/include LIB_DIR=/path/to/lib'

   or cloned the GitHub git2r repository into a directory:
   R CMD INSTALL git2r/ --configure-vars='INCLUDE_DIR=/path/to/include LIB_DIR=/path/to/lib'

   or download and install git2r in R using
   install.packages('git2r', type='source', configure.vars='LIB_DIR=-L/path/to/libs INCLUDE_DIR=-I/path/to/headers')

  -----------------------------------------------------------------------


ERROR: configuration failed for package ‘git2r’
* removing ‘/__w/_temp/Library/git2r’
```
leading to:
```
ERROR: dependency ‘git2r’ is not available for package ‘atime’
* removing ‘/__w/_temp/Library/atime’

The downloaded source packages are in
	‘/tmp/RtmpoQfIF7/downloaded_packages’
Warning messages:
1: In install.packages(c("atime", "ggplot2", "directlabels")) :
  installation of package ‘git2r’ had non-zero exit status
2: In install.packages(c("atime", "ggplot2", "directlabels")) :
  installation of package ‘atime’ had non-zero exit status
Error: Error in loadNamespace(x) : there is no package called ‘atime’
Calls: loadNamespace -> withRestarts -> withOneRestart -> doWithOneRestart
Execution halted
Error: Process completed with exit code 1.
```
Keeping in mind what the log above states, my two cents would be to first investigate if the cause is potentially:
- Version mismatch (considering `libgit2` is indeed outdated or out of sync with `git2r` req)
- Missing proper path spec
- Changes to libraries/configurations

**Comment by Anirban166** [comment]
```sh
Run sudo add-apt-repository ppa:git-core/ppa
  sudo add-apt-repository ppa:git-core/ppa
  sudo apt-get update -y
  sudo apt-get install -y libgit2-dev
  dpkg -L libgit2-dev
  pkg-config --modversion libgit2
  shell: bash --noprofile --norc -e -o pipefail {0}
  env:
    GITHUB_PAT: ***
    repo_token: ***
    START_SETUP_TIME: 1730495637
    R_LIBS_USER: /__w/_temp/Library
    TZ: UTC
    _R_CHECK_SYSTEM_CLOCK_: FALSE
    NOT_CRAN: true
    RSPM: https://packagemanager.posit.co/cran/__linux__/focal/latest
    RENV_CONFIG_REPOS_OVERRIDE: https://packagemanager.posit.co/cran/__linux__/focal/latest
 The most current stable version of Git for Ubuntu.

For release candidates, go to https://launchpad.net/~git-core/+archive/candidate .
 More info: https://launchpad.net/~git-core/+archive/ubuntu/ppa
Hit:1 http://security.ubuntu.com/ubuntu focal-security InRelease
Hit:2 https://deb.nodesource.com/node_16.x nodistro InRelease
Hit:3 http://archive.ubuntu.com/ubuntu focal InRelease
Hit:4 http://ppa.launchpad.net/git-core/ppa/ubuntu focal InRelease
Hit:5 http://archive.ubuntu.com/ubuntu focal-updates InRelease
Hit:6 http://ppa.launchpad.net/longsleep/golang-backports/ubuntu focal InRelease
Hit:7 http://archive.ubuntu.com/ubuntu focal-backports InRelease
Get:8 https://s3-us-east-2.amazonaws.com/dvc-s3-repo/deb stable InRelease [2,679 B]
Ign:8 https://s3-us-east-2.amazonaws.com/dvc-s3-repo/deb stable InRelease
Fetched 2,679 B in 1s (3,518 B/s)
Reading package lists...
W: GPG error: https://s3-us-east-2.amazonaws.com/dvc-s3-repo/deb stable InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY EFFAF4443CE9AD84
Hit:1 http://archive.ubuntu.com/ubuntu focal InRelease
Hit:2 http://archive.ubuntu.com/ubuntu focal-updates InRelease
Hit:3 http://archive.ubuntu.com/ubuntu focal-backports InRelease
Hit:4 https://deb.nodesource.com/node_16.x nodistro InRelease
Hit:5 http://security.ubuntu.com/ubuntu focal-security InRelease
Get:6 https://s3-us-east-2.amazonaws.com/dvc-s3-repo/deb stable InRelease [2,679 B]
Hit:7 http://ppa.launchpad.net/git-core/ppa/ubuntu focal InRelease
Hit:8 http://ppa.launchpad.net/longsleep/golang-backports/ubuntu focal InRelease
Ign:6 https://s3-us-east-2.amazonaws.com/dvc-s3-repo/deb stable InRelease
Fetched 2,679 B in 1s (3,587 B/s)
Reading package lists...
W: GPG error: https://s3-us-east-2.amazonaws.com/dvc-s3-repo/deb stable InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY EFFAF4443CE9AD84
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  libgcrypt20-dev libgit2-28 libgpg-error-dev libhttp-parser-dev
  libhttp-parser2.9 libmbedcrypto3 libmbedtls-dev libmbedtls12 libmbedx509-0
  libssh2-1 libssh2-1-dev
Suggested packages:
  libgcrypt20-doc libmbedtls-doc
The following NEW packages will be installed:
  libgcrypt20-dev libgit2-28 libgit2-dev libgpg-error-dev libhttp-parser-dev
  libhttp-parser2.9 libmbedcrypto3 libmbedtls-dev libmbedtls12 libmbedx509-0
  libssh2-1 libssh2-1-dev
0 upgraded, 12 newly installed, 0 to remove and 5 not upgraded.
Need to get 2,745 kB of archives.
After this operation, 11.4 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu focal/main amd64 libgpg-error-dev amd64 1.37-1 [109 kB]
Get:2 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libgcrypt20-dev amd64 1.8.5-5ubuntu1.1 [471 kB]
Get:3 http://archive.ubuntu.com/ubuntu focal/universe amd64 libhttp-parser2.9 amd64 2.9.2-2 [21.8 kB]
Get:4 http://archive.ubuntu.com/ubuntu focal/universe amd64 libmbedcrypto3 amd64 2.16.4-1ubuntu2 [150 kB]
Get:5 http://archive.ubuntu.com/ubuntu focal/universe amd64 libmbedx509-0 amd64 2.16.4-1ubuntu2 [42.3 kB]
Get:6 http://archive.ubuntu.com/ubuntu focal/universe amd64 libmbedtls12 amd64 2.16.4-1ubuntu2 [71.8 kB]
Get:7 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 libssh2-1 amd64 1.8.0-2.1ubuntu0.1 [75.5 kB]
Get:8 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 libgit2-28 amd64 0.28.4+dfsg.1-2ubuntu0.1 [403 kB]
Get:9 http://archive.ubuntu.com/ubuntu focal/universe amd64 libmbedtls-dev amd64 2.16.4-1ubuntu2 [444 kB]
Get:10 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 libssh2-1-dev amd64 1.8.0-2.1ubuntu0.1 [244 kB]
Get:11 http://archive.ubuntu.com/ubuntu focal/universe amd64 libhttp-parser-dev amd64 2.9.2-2 [20.2 kB]
Get:12 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 libgit2-dev amd64 0.28.4+dfsg.1-2ubuntu0.1 [692 kB]
debconf: unable to initialize frontend: Dialog
debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (This frontend requires a controlling tty.)
debconf: falling back to frontend: Teletype
dpkg-preconfigure: unable to re-open stdin: 
Fetched 2,745 kB in 0s (15.0 MB/s)
Selecting previously unselected package libgpg-error-dev.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 55554 files and directories currently installed.)
Preparing to unpack .../00-libgpg-error-dev_1.37-1_amd64.deb ...
Unpacking libgpg-error-dev (1.37-1) ...
Selecting previously unselected package libgcrypt20-dev.
Preparing to unpack .../01-libgcrypt20-dev_1.8.5-5ubuntu1.1_amd64.deb ...
Unpacking libgcrypt20-dev (1.8.5-5ubuntu1.1) ...
Selecting previously unselected package libhttp-parser2.9:amd64.
Preparing to unpack .../02-libhttp-parser2.9_2.9.2-2_amd64.deb ...
Unpacking libhttp-parser2.9:amd64 (2.9.2-2) ...
Selecting previously unselected package libmbedcrypto3:amd64.
Preparing to unpack .../03-libmbedcrypto3_2.16.4-1ubuntu2_amd64.deb ...
Unpacking libmbedcrypto3:amd64 (2.16.4-1ubuntu2) ...
Selecting previously unselected package libmbedx509-0:amd64.
Preparing to unpack .../04-libmbedx509-0_2.16.4-1ubuntu2_amd64.deb ...
Unpacking libmbedx509-0:amd64 (2.16.4-1ubuntu2) ...
Selecting previously unselected package libmbedtls12:amd64.
Preparing to unpack .../05-libmbedtls12_2.16.4-1ubuntu2_amd64.deb ...
Unpacking libmbedtls12:amd64 (2.16.4-1ubuntu2) ...
Selecting previously unselected package libssh2-1:amd64.
Preparing to unpack .../06-libssh2-1_1.8.0-2.1ubuntu0.1_amd64.deb ...
Unpacking libssh2-1:amd64 (1.8.0-2.1ubuntu0.1) ...
Selecting previously unselected package libgit2-28:amd64.
Preparing to unpack .../07-libgit2-28_0.28.4+dfsg.1-2ubuntu0.1_amd64.deb ...
Unpacking libgit2-28:amd64 (0.28.4+dfsg.1-2ubuntu0.1) ...
Selecting previously unselected package libmbedtls-dev:amd64.
Preparing to unpack .../08-libmbedtls-dev_2.16.4-1ubuntu2_amd64.deb ...
Unpacking libmbedtls-dev:amd64 (2.16.4-1ubuntu2) ...
Selecting previously unselected package libssh2-1-dev:amd64.
Preparing to unpack .../09-libssh2-1-dev_1.8.0-2.1ubuntu0.1_amd64.deb ...
Unpacking libssh2-1-dev:amd64 (1.8.0-2.1ubuntu0.1) ...
Selecting previously unselected package libhttp-parser-dev.
Preparing to unpack .../10-libhttp-parser-dev_2.9.2-2_amd64.deb ...
Unpacking libhttp-parser-dev (2.9.2-2) ...
Selecting previously unselected package libgit2-dev:amd64.
Preparing to unpack .../11-libgit2-dev_0.28.4+dfsg.1-2ubuntu0.1_amd64.deb ...
Unpacking libgit2-dev:amd64 (0.28.4+dfsg.1-2ubuntu0.1) ...
Setting up libgpg-error-dev (1.37-1) ...
Setting up libmbedcrypto3:amd64 (2.16.4-1ubuntu2) ...
Setting up libssh2-1:amd64 (1.8.0-2.1ubuntu0.1) ...
Setting up libhttp-parser2.9:amd64 (2.9.2-2) ...
Setting up libhttp-parser-dev (2.9.2-2) ...
Setting up libmbedx509-0:amd64 (2.16.4-1ubuntu2) ...
Setting up libmbedtls12:amd64 (2.16.4-1ubuntu2) ...
Setting up libmbedtls-dev:amd64 (2.16.4-1ubuntu2) ...
Setting up libgcrypt20-dev (1.8.5-5ubuntu1.1) ...
Setting up libssh2-1-dev:amd64 (1.8.0-2.1ubuntu0.1) ...
Setting up libgit2-28:amd64 (0.28.4+dfsg.1-2ubuntu0.1) ...
Setting up libgit2-dev:amd64 (0.28.4+dfsg.1-2ubuntu0.1) ...
Processing triggers for man-db (2.9.1-1) ...
Processing triggers for libc-bin (2.31-0ubuntu9.16) ...
/.
/usr
/usr/include
/usr/include/git2
/usr/include/git2/annotated_commit.h
/usr/include/git2/apply.h
/usr/include/git2/attr.h
/usr/include/git2/blame.h
/usr/include/git2/blob.h
/usr/include/git2/branch.h
/usr/include/git2/buffer.h
/usr/include/git2/checkout.h
/usr/include/git2/cherrypick.h
/usr/include/git2/clone.h
/usr/include/git2/commit.h
/usr/include/git2/common.h
/usr/include/git2/config.h
/usr/include/git2/cred_helpers.h
/usr/include/git2/deprecated.h
/usr/include/git2/describe.h
/usr/include/git2/diff.h
/usr/include/git2/errors.h
/usr/include/git2/filter.h
/usr/include/git2/global.h
/usr/include/git2/graph.h
/usr/include/git2/ignore.h
/usr/include/git2/index.h
/usr/include/git2/indexer.h
/usr/include/git2/inttypes.h
/usr/include/git2/mailmap.h
/usr/include/git2/merge.h
/usr/include/git2/message.h
/usr/include/git2/net.h
/usr/include/git2/notes.h
/usr/include/git2/object.h
/usr/include/git2/odb.h
/usr/include/git2/odb_backend.h
/usr/include/git2/oid.h
/usr/include/git2/oidarray.h
/usr/include/git2/pack.h
/usr/include/git2/patch.h
/usr/include/git2/pathspec.h
/usr/include/git2/proxy.h
/usr/include/git2/rebase.h
/usr/include/git2/refdb.h
/usr/include/git2/reflog.h
/usr/include/git2/refs.h
/usr/include/git2/refspec.h
/usr/include/git2/remote.h
/usr/include/git2/repository.h
/usr/include/git2/reset.h
/usr/include/git2/revert.h
/usr/include/git2/revparse.h
/usr/include/git2/revwalk.h
/usr/include/git2/signature.h
/usr/include/git2/stash.h
/usr/include/git2/status.h
/usr/include/git2/stdint.h
/usr/include/git2/strarray.h
/usr/include/git2/submodule.h
/usr/include/git2/sys
/usr/include/git2/sys/alloc.h
/usr/include/git2/sys/commit.h
/usr/include/git2/sys/config.h
/usr/include/git2/sys/diff.h
/usr/include/git2/sys/filter.h
/usr/include/git2/sys/hashsig.h
/usr/include/git2/sys/index.h
/usr/include/git2/sys/mempack.h
/usr/include/git2/sys/merge.h
/usr/include/git2/sys/odb_backend.h
/usr/include/git2/sys/openssl.h
/usr/include/git2/sys/path.h
/usr/include/git2/sys/refdb_backend.h
/usr/include/git2/sys/reflog.h
/usr/include/git2/sys/refs.h
/usr/include/git2/sys/repository.h
/usr/include/git2/sys/stream.h
/usr/include/git2/sys/time.h
/usr/include/git2/sys/transport.h
/usr/include/git2/tag.h
/usr/include/git2/trace.h
/usr/include/git2/transaction.h
/usr/include/git2/transport.h
/usr/include/git2/tree.h
/usr/include/git2/types.h
/usr/include/git2/version.h
/usr/include/git2/worktree.h
/usr/include/git2.h
/usr/lib
/usr/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu/libgit2.a
/usr/lib/x86_64-linux-gnu/pkgconfig
/usr/lib/x86_64-linux-gnu/pkgconfig/libgit2.pc
/usr/share
/usr/share/doc
/usr/share/doc/libgit2-dev
/usr/share/doc/libgit2-dev/checkout-internals.md.gz
/usr/share/doc/libgit2-dev/code_of_conduct.md
/usr/share/doc/libgit2-dev/contributing.md.gz
/usr/share/doc/libgit2-dev/conventions.md.gz
/usr/share/doc/libgit2-dev/copyright
/usr/share/doc/libgit2-dev/diff-internals.md.gz
/usr/share/doc/libgit2-dev/differences-from-git.md
/usr/share/doc/libgit2-dev/error-handling.md.gz
/usr/share/doc/libgit2-dev/examples
/usr/share/doc/libgit2-dev/examples/CMakeLists.txt
/usr/share/doc/libgit2-dev/examples/COPYING
/usr/share/doc/libgit2-dev/examples/README.md
/usr/share/doc/libgit2-dev/examples/add.c
/usr/share/doc/libgit2-dev/examples/blame.c
/usr/share/doc/libgit2-dev/examples/cat-file.c
/usr/share/doc/libgit2-dev/examples/checkout.c
/usr/share/doc/libgit2-dev/examples/common.c
/usr/share/doc/libgit2-dev/examples/common.h
/usr/share/doc/libgit2-dev/examples/describe.c
/usr/share/doc/libgit2-dev/examples/diff.c
/usr/share/doc/libgit2-dev/examples/for-each-ref.c
/usr/share/doc/libgit2-dev/examples/general.c
/usr/share/doc/libgit2-dev/examples/init.c
/usr/share/doc/libgit2-dev/examples/log.c
/usr/share/doc/libgit2-dev/examples/ls-files.c
/usr/share/doc/libgit2-dev/examples/merge.c
/usr/share/doc/libgit2-dev/examples/network
/usr/share/doc/libgit2-dev/examples/network/Makefile
/usr/share/doc/libgit2-dev/examples/network/clone.c
/usr/share/doc/libgit2-dev/examples/network/common.c
/usr/share/doc/libgit2-dev/examples/network/common.h
/usr/share/doc/libgit2-dev/examples/network/fetch.c
/usr/share/doc/libgit2-dev/examples/network/git2.c
/usr/share/doc/libgit2-dev/examples/network/index-pack.c
/usr/share/doc/libgit2-dev/examples/network/ls-remote.c
/usr/share/doc/libgit2-dev/examples/remote.c
/usr/share/doc/libgit2-dev/examples/rev-list.c
/usr/share/doc/libgit2-dev/examples/rev-parse.c
/usr/share/doc/libgit2-dev/examples/showindex.c
/usr/share/doc/libgit2-dev/examples/status.c
/usr/share/doc/libgit2-dev/examples/tag.c
/usr/share/doc/libgit2-dev/examples/test
/usr/share/doc/libgit2-dev/examples/test/test-rev-list.sh
/usr/share/doc/libgit2-dev/fuzzing.md
/usr/share/doc/libgit2-dev/merge-df_conflicts.txt
/usr/share/doc/libgit2-dev/projects.md.gz
/usr/share/doc/libgit2-dev/release.md.gz
/usr/share/doc/libgit2-dev/threading.md.gz
/usr/share/doc/libgit2-dev/troubleshooting.md
/usr/lib/x86_64-linux-gnu/libgit2.so
/usr/share/doc/libgit2-dev/changelog.Debian.gz
0.28.4
```

**Comment by Anirban166** [comment]
Alright it's been fixed now (since v.1.4.0). The problem was version incompatibility (which was my first guess), as an older version was being installed even after updating packages with `apt` or using the `git-core` PPA:
```sh
pkg-config --modversion libgit2
```
```
0.28.4
```
File paths:
```sh
find /usr -name 'libgit2*' -type d
```
```
/usr/share/doc/libgit2-28
/usr/share/doc/libgit2-dev
```
The current approach fetches and installs `libgit2` from source, ensuring that we have the latest version (manually specified, currently set to v1.6.4). It builds the library with HTTPS support (enabled by `libssl-dev`) and installs it in `/usr` so other tools and libraries on the system can easily access it.
